torch/distributed/elastic/multiprocessing/api.py

```diff
@@ -670,9 +670,14 @@ def _poll(self) -> Optional[RunProcsResult]:
             if self._is_done():
                 # we should ALWAYS have ALL the return values when all the processes are done
                 self._worker_finished_event.set()
-                # Wait untill all processes are finished. At this point workers finished executing
-                # user function
-                self._pc.join()
+
+                # At this point workers finished running the user function
+                # But the child process might still have not exited. Wait for them.
+                # pc.join() blocks [forever] until "a" proc exits. Loop until all of them exits.
+                all_procs_exited = self._pc.join()
+                while not all_procs_exited:
+                    logger.debug("entrypoint fn finished, waiting for all child procs to exit...")
```

Comment:
I assume this should have `all_procs_exited = self._pc.join()` within the while loop? Can we add a test for this behavior?